src/apme_engine/graph/variable_helpers.py

```
@@ -0,0 +1,497 @@
+        node_id: Task or handler node id.
+        play_scope: Precomputed play-scoped node IDs.
+
+    Returns:
+        True when any in-scope ancestor path resolves ``no_log`` to true.
+    """
+    seen_paths: set[tuple[str, ...]] = set()
+
+    def walk(current_id: str, path: tuple[str, ...]) -> bool:
+        if current_id in path:
+            return False
+        next_path = path + (current_id,)
```

Ruff (0.16.2)

[warning] Consider `(*path, current_id)` instead of concatenation

Replace with `(*path, current_id)`

(RUF005)